gadget/filesystemimage.go

```diff
@@ -33,7 +33,7 @@ type MkfsFunc func(imgFile, label, contentsRootDir string) error
 var (
 	mkfsHandlers = map[string]MkfsFunc{
 		"vfat": MkfsVfat,
-		"ext4": MkfsExt4,
+		"ext4": MkfsExt4WithoutMetadataChecksum,
```

Comment:
Does this mean for any gadget update we need to do which requires mkfs we will now run the uc18+ mkfs ?

Reply:
I tried to be conservative and only filesystems created during the uc20 installation will have metadata checksum. `MkfsExt4WithoutMetadataChecksum` is what we used to call `MkfsExt4` (I used the plain name for the command without extra options, and the special name for the command with extra options to disable `metadata_csum`).

Reply:
Note that the old e2fsprogs require `metadata_csum` to be disabled (and replaced by `uninit_bg`), and the 600P requires it to be enabled.

From ext4(5), `metadata_csum` and `uninit_bg` are incompatible and `metadata_csum` will be used preferentially.

Reply:
Aha, sorry. I see  how much this is all hanging together. Maybe a comment here like:
```
// disable some ext4 features by default to support UC16 (16.04) based devices.
```

Reply:
afaiu this code is never used by anything atm, in particular also not gadget updates. current gadget updates don't create filesystems